src/backend/dx12/src/device.rs

```diff
@@ -2334,18 +2334,8 @@ impl d::Device<B> for Device {
                                                .map(|desc| *desc.borrow())
                                                .collect::<Vec<_>>();
         for desc in &descriptor_pools {
```

Comment:
Based on commit: https://github.com/gfx-rs/gfx/commit/1f9b40e4 both `num_samplers` and `num_srv_cbv_uav` must be incremented, regardless on what type of DescriptorType is passed in.

Reply:
Not sure how got went from the commit to the idea. The commit is for Metal backend specifically, and in general I don't see why we'd want to increase the sampler count for descriptors that have nothing to do with samplers.

Reply:
Yes sorry for the late reaction. We spoke about this in Gitter and what I think the best solution would be is: check if `num_samplers` is equal to 0, if that is so I want to return a None, else I want to return the heap_sampler in the Some. 

Same goes for `num_srv_cbv_uav`.

Is this a correct solution?

Reply:
So about the solution, maybe something like this (bear with me, written from my head and without testing):

`fn get_descriptor_sampler(number: Option<?>) -> Option<?> {
    match numer {
        0 => None,
        _ => {
                    let mut heap_sampler = self
                       .heap_sampler
                       .lock()
                       .unwrap();

                    let range = heap_sampler
                       .range_allocator
                       .allocate_range(num_samplers as _)
                       .unwrap(); // TODO: error/resize

                    Some(n::DescriptorHeapSlice {
                        heap: heap_sampler.raw.clone(),
                        handle_size: heap_sampler.handle_size as _,
                        range_allocator: RangeAllocator::new(range),
                        start: heap_sampler.start,
                    })
        }
    }
}`

I am not sure about the return type of the option.

Reply:
So you'd want to have both `heap_sampler` and `heap_srv_cbv_uav` to be stored as `Option<>`, based on whether or not the appropriate count was 0.

Reply:
Yes, that is right.